### PR TITLE
docs: Move router tutorial (toh) from router.md to new file

### PR DESCRIPTION
**Pull request by aikithoughts**
In an effort to make angular documentation easier for users to read,
we are moving the router tutorial currently in router.md to a new file.
To support this change, we have done the following:

* Update files to fix any broken links caused by moving the file
* Updated the new file to follow tutorial guidelines
* Add the new file to the table of  contents under, Tutorials.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Current tutorial is buried in the Router.md topic.

Issue Number: N/A

## What is the new behavior?
Move the tutorial to its own file and add it to the left navigation.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by atscott** [comment]
Hey Dave, wouldn't this conflict with #37405?

File: packages/router/src/events.ts [hunk 1]
```diff
@@ -127,7 +127,7 @@ export class NavigationEnd extends RouterEvent {
 /**
  * An event triggered when a navigation is canceled, directly or indirectly.
  *
- * This can happen when a [route guard](guide/router#milestone-5-route-guards)
+ * This can happen when a [route guard](guide/router-tutorial-toh#milestone-5-route-guards)
```
Comment (jbogarthyde) [inline packages/router/src/events.ts:130]: This should point to the guide -- `[route guard](guide/router#guards)`

**Review by atscott — APPROVED** [review]

**Comment by jbogarthyde** [comment]
This definitely conflicts with https://github.com/angular/angular/pull/37405

File: packages/router/src/events.ts (hunk 1 above)
Comment (aikidave) [reply packages/router/src/events.ts:130]: I will address that work in a separate PR. Thanks!

**Comment by jelbourn** [comment]
@aikidave looks like there's a lint problem with the commit message

**Review by kapunahelewong — APPROVED** [review]
LGTM!
Reviewed-for: global-docs-approvers

**Comment by aikidave** [comment]
I think this is okay to review now, @jelbourn .

**Review by jelbourn — APPROVED** [review]
LGTM, using global approvers since only Igor has the specific codeowners approval

Reviewed-for: global-approvers

**Pull request closed without merging**